lib/tasks/add_browse_users.rake

```diff
@@ -0,0 +1,39 @@
+# rake goodcity:add_user
+namespace :goodcity do
+  desc "Add new browse user 'MOBILE=+85261111111 FIRST_NAME=Steve LAST_NAME=Jobs ORGANISATION=Apple cap production invoke:rake TASK=goodcity:add_user'  "
+  task add_user: :environment do
+    get_user_details
+  end
+
+  def get_user_details
+    f_name = ENV['FIRST_NAME']
+    l_name = ENV['LAST_NAME']
+    mobile = ENV['MOBILE']
+    o_name = ENV['ORGANISATION']
+    if (f_name && l_name && mobile && o_name)
+      user = create_new_user(f_name, l_name, mobile)
+      add_user_to_organisation(user, o_name)
+    else
+      incorrect_command_prompt
+    end
+  end
+
+  def incorrect_command_prompt
+    puts "Incorrect command:\tEnter command in one of the following format:"
+    puts "\t MOBILE=+85261111111 FIRST_NAME=Steve LAST_NAME=Jobs ORGANISATION=Apple cap production invoke:rake TASK=goodcity:add_user"
+    puts "\t rake goodcity:add_user MOBILE=+85261111111 FIRST_NAME=Steve LAST_NAME=Jobs ORGANISATION=Apple"
+  end
+
+  def add_user_to_organisation(user, o_name)
+    if(org = Organisation.find_by_name_en(o_name))
+      organisation.users << user if(user.save!)
```

Comment:
Code dies at this point because `organisation` variable doesn't exist. A test would have picked this up immediately. For rake tasks, I really would like to see rake as a wrapper to a `lib/` class that performs the actions required. That way, it's easy to write tests to prove it works, easy to call via some other method (e.g. rails console or script) and generally much more flexible.

Suggest the following:
* Move all methods to lib/goodcity/add_user.rb `AddUser` class.
* AddUser class handles user creation and assignment to correct organisation. It's essentially a utility class.
* Rake passes the ENV vars through to the AddUser class
* AddUser is well tested with rspecs.